src/tests/java/picard/analysis/directed/CollectTargetedMetricsTest.java

```diff
@@ -0,0 +1,159 @@
+public class CollectTargetedMetricsTest extends CommandLineProgramTest {
+    private final String referenceFile = "testdata/picard/quality/chrM.reference.fasta";
+    private final static File TEST_DIR = new File("testdata/picard/sam/CollectGcBiasMetrics/");
+    private final String emptyIntervals = "testdata/picard/quality/chrM.empty.interval_list";
+    private final String singleIntervals = "testdata/picard/quality/chrM.single.interval_list";
+    private final File dict = new File(TEST_DIR, "Mheader.dict");
+    private File tempSamFile;
+    private File outfile;
+    private File perTargetOutfile;
+
+    private final static String sample = "TestSample1";
+    private final static String readGroupId = "TestReadGroup1";
+    private final static String platform = "ILLUMINA";
+    private final static String library = "TestLibrary1";
+    private final static int numReads = 40000;
+
+    @Override
+    public String getCommandLineProgramName() {
+        return CollectTargetedPcrMetrics.class.getSimpleName();
+    }
+
+    //create a samfile with 40000 reads for testing whether a cap is found.
+    @BeforeTest
+    void setupBuilder() throws IOException {
+        final String readName = "TESTBARCODE";
+
+        //Create Sam Files
+        tempSamFile = File.createTempFile("CollectTargetedMetrics", ".bam", TEST_DIR);
+        final File tempSamFileUnsorted = File.createTempFile("CollectTargetedMetrics", ".bam", TEST_DIR);
+        tempSamFileUnsorted.deleteOnExit();
+        tempSamFile.deleteOnExit();
+        final SAMFileHeader header = new SAMFileHeader();
+
+        //Check that dictionary file is readable and then set header dictionary
+        try {
+            header.setSequenceDictionary(SAMSequenceDictionaryExtractor.extractDictionary(dict));
+            header.setSortOrder(SAMFileHeader.SortOrder.unsorted);
+        } catch (final SAMException e) {
+            e.printStackTrace();
+        }
+
+        //Set readGroupRecord
+        final SAMReadGroupRecord readGroupRecord = new SAMReadGroupRecord(readGroupId);
+        readGroupRecord.setSample(sample);
+        readGroupRecord.setPlatform(platform);
+        readGroupRecord.setLibrary(library);
+        readGroupRecord.setPlatformUnit(readGroupId);
+        header.addReadGroup(readGroupRecord);
+
+        //Add to setBuilder
+        final SAMRecordSetBuilder setBuilder = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+        setBuilder.setReadGroup(readGroupRecord);
+        setBuilder.setUseNmFlag(true);
+        setBuilder.setHeader(header);
+
+        //Read settings
+        final String separator = ":";
+        final int ID = 1;
+        final int max = 15000;
+        final int min = 1;
+        final Random rg = new Random(5);
+
+        for (int i = 0; i < numReads; i++) {
+            final int start = rg.nextInt(max) + min;
+            final String newReadName = readName + separator + ID + separator + i;
+            final int LENGTH = 99;
```

Comment:
extract Length from loop. capitalization is fitting for a static variable, so put it all the way up in the class.